lib/token-assert.js

```diff
@@ -31,19 +31,23 @@ TokenAssert.prototype.whitespaceBetween = function(options) {
         if (nextToken.loc.start.line === token.loc.end.line &&
             (nextToken.loc.start.column - token.loc.end.column) !== spaces
         ) {
+            nextToken.whitespaceBefore = new Array(spaces + 1).join(' ');
```

Comment:
Well, think of it this way. If we both modify whitespaceBefore and use it for checking, then we avoid a lot of cases where two full runs would be needed.

Imagine a "requireNewline" rule and a "requireSpace" rule that both report on a `{` at the end of a line...

If we had rules check the whitespaceBefore property, then a fix for requireNewline would make requireSpace ignore.

If we don't, then 2 passes must happen, one fixing the spacing, while the second pass finally moved it to the right spot on the next line.